### PR TITLE
Add json support for BidRequest $.user.consent, $.regs.gpp, $.regs.gpp_sid, $.regs.gdpr, $.regs.us_privacy

### PR DESCRIPTION
**Pull request by vaa25**

### CODE REVIEW

File: openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonReader.java [hunk 1]
```diff
@@ -256,6 +256,20 @@ protected void readRegsField(JsonParser par, Regs.Builder reg, String fieldName)
       case "coppa":
         reg.setCoppa(par.getValueAsBoolean());
         break;
+      case "gpp":
+        reg.setGpp(par.getText());
+        break;
+      case "gppsid":
```
Comment (dmytro-babichev) [inline openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonReader.java]: gpp_sid

File: openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonReader.java [hunk 2]
```diff
@@ -256,6 +256,20 @@ protected void readRegsField(JsonParser par, Regs.Builder reg, String fieldName)
...
+        for (startArray(par); endArray(par); par.nextToken()) {
+          reg.addGppSid(par.getIntValue());
+        }
+        break;
+      case "gdpr":
+        reg.setGdpr(par.getValueAsBoolean());
+        break;
+      case "usprivacy":
```
Comment (dmytro-babichev) [inline openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonReader.java]: us_privacy

File: openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java [hunk 3]
```diff
@@ -1101,6 +1104,16 @@ protected void writeRegsFields(Regs regs, JsonGenerator gen) throws IOException
     if (regs.hasCoppa()) {
       writeIntBoolField("coppa", regs.getCoppa(), gen);
     }
+    if (regs.hasGpp()) {
+      gen.writeStringField("gpp", regs.getGpp());
+    }
+    writeInts("gppsid", regs.getGppSidList(), gen);
```
Comment (dmytro-babichev) [inline openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java]: gpp_sid

File: openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java [hunk 4]
```diff
@@ -1101,6 +1104,16 @@ protected void writeRegsFields(Regs regs, JsonGenerator gen) throws IOException
...
+    if (regs.hasGdpr()) {
+      writeIntBoolField("gdpr", regs.getGdpr(), gen);
+    }
+    if (regs.hasUsPrivacy()) {
+      gen.writeStringField("usprivacy", regs.getUsPrivacy());
```
Comment (dmytro-babichev) [inline openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java]: us_privacy

**Review by dmytro-babichev — CHANGES_REQUESTED** [review]
those field names are a part of the standards OpenRTB 2.6 and GPPv1, we must follow it

**Review by dmytro-babichev — APPROVED** [review]

**Review by simontrasler — APPROVED** [review]
LGTM.

**Pull request merged**